tensorflow/core/common_runtime/accumulate_n_optimizer.cc

```diff
@@ -0,0 +1,204 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/common_runtime/graph_optimizer.h"
+
+#include "tensorflow/core/common_runtime/constant_folding.h"
+#include "tensorflow/core/common_runtime/function.h"
+#include "tensorflow/core/common_runtime/optimization_registry.h"
+#include "tensorflow/core/graph/algorithm.h"
+#include "tensorflow/core/graph/node_builder.h"
+#include "tensorflow/core/graph/optimizer_cse.h"
+
+namespace tensorflow {
+namespace {
+
+Tensor make_zeros(const DataType& dtype, const TensorShapeProto& shape) {
+  Tensor tensor(dtype, TensorShape(shape));
+
+  // Conveniently, all numeric data types have 0x0 == zero.  Otherwise we would
+  // need a giant switch statement here.
+  memset(const_cast<char*>(tensor.tensor_data().data()), 0,
+         tensor.tensor_data().size());
+
+  return tensor;
+}
+
+// TEMPORARY debugging code
```

Comment:
Remove comment and debugging code

Reply:
Ugh, forgot about that. Those two lines are removed now.

Reply:
Thanks! Still waiting on the other change (related to forward compatibility)